frappe/rate_limiter.py

```diff
@@ -111,11 +111,8 @@ def ratelimit_decorator(fun):
 		@wraps(fun)
 		def wrapper(*args, **kwargs):
 			# Do not apply rate limits if method is not opted to check
-			if (
-				methods != "ALL"
-				and frappe.request
-				and frappe.request.method
-				and frappe.request.method.upper() not in methods
+			if not frappe.request or (
+				methods != "ALL" and frappe.request.method and frappe.request.method.upper() not in methods
 			):
 				return frappe.call(fun, **frappe.form_dict or kwargs)
```

Comment:
form_dict wont exist. in case there's no frappe.request

Reply:
![image](https://github.com/frappe/frappe/assets/16315650/21e2cda3-14c4-48b6-b3de-39b75a52659f)

Reply:
but using `form_dict` is logically incorrect, rate limiter isn't supposed to do that magic

Reply:
Yeah I agree. I tried to remove this while ago and forgot why it was done this way.